cmd/cogito/main_test.go

```diff
@@ -134,6 +135,46 @@ func TestRunPutSuccessIntegration(t *testing.T) {
 		`level=INFO msg="state posted successfully to chat" name=cogito.put name=gChat state=error`))
 }
 
+func TestRunPutGhAppSuccessIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("Skipping integration test (reason: -short)")
+	}
+
+	gitHubCfg := testhelp.GitHubSecretsOrFail(t)
+	ghAppInstallationID, err := strconv.ParseInt(gitHubCfg.GhAppInstallationID, 10, 64)
+	assert.NilError(t, err)
+
+	stdin := bytes.NewReader(testhelp.ToJSON(t, cogito.PutRequest{
+		Source: cogito.Source{
+			Owner: gitHubCfg.Owner,
+			Repo:  gitHubCfg.Repo,
+			GitHubApp: github.GitHubApp{
+				ClientId:       gitHubCfg.GhAppClientID,
+				InstallationId: ghAppInstallationID,
+				PrivateKey:     gitHubCfg.GhAppPrivateKey,
+			},
+			LogLevel: "debug",
+		},
+		Params: cogito.PutParams{State: cogito.StateError},
+	}))
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	inputDir := testhelp.MakeGitRepoFromTestdata(t, "../../cogito/testdata/one-repo/a-repo",
+		testhelp.HttpsRemote(github.GhDefaultHostname, gitHubCfg.Owner, gitHubCfg.Repo), gitHubCfg.SHA,
+		"ref: refs/heads/a-branch-FIXME")
+	t.Setenv("BUILD_JOB_NAME", "TestRunPutGhAppSuccessIntegration")
+	t.Setenv("ATC_EXTERNAL_URL", "https://cogito.invalid")
+	t.Setenv("BUILD_PIPELINE_NAME", "the-test-pipeline")
+	t.Setenv("BUILD_TEAM_NAME", "the-test-team")
+	t.Setenv("BUILD_NAME", "42")
+
+	err = mainErr(stdin, &stdout, &stderr, []string{"out", inputDir})
+
+	assert.NilError(t, err, "\nstdout:\n%s\nstderr:\n%s", stdout.String(), stderr.String())
+	assert.Assert(t, cmp.Contains(stderr.String(),
+		`level=INFO msg="commit status posted successfully" name=cogito.put name=ghCommitStatus state=error`))
```

Comment:
Mhh.
Are we peeking in the logs like this in the other tests?

Reply:
In other integration tests... I copied and pasted it :) I think it was written by you actually :)